clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -5151,6 +5151,14 @@ bool TokenAnnotator::mustBreakBefore(const AnnotatedLine &Line,
     return true;
   if (Left.IsUnterminatedLiteral)
     return true;
+  // FIXME: Breaking after newlines seems useful in general. Turn this into an
+  // option and recognize more cases like endl etc, and break independent of
+  // what comes after operator lessless.
+  if (Right.is(tok::lessless) && Right.Next &&
+      Right.Next->is(tok::string_literal) && Left.is(tok::string_literal) &&
+      Left.TokenText.ends_with("\\n\"")) {
```

Comment:
We can also handle `\n` and `endl` now. For example:
```suggestion
      Right.Next->is(tok::string_literal) &&
      ((Left.is(tok::string_literal) && Left.TokenText.ends_with("\\n\"")) ||
       Left.TokenText == "'\\n'" || Left.TokenText == "endl")) {
```

Reply:
> We can also handle \n and endl now. For example:

yes, but that'll be "new" behavior, and even if we do that, i'd rather do that in a separate patch that can be reviewed/reverted on it's own.

as mentioned, this is mostly to restore some of the "incidental" behavior that was relied on by codebases in the wild. so that changes in the new clang-format release will be minimal.